test/test_checkers.py

```diff
@@ -145,12 +145,54 @@ class MyFakeChecker(Checker):
         VENDOR_PRODUCT = [("my", "checker")]
         IGNORE_PATTERNS = [r"mychecker-5.6"]
 
-    string = "Some other lines. \n Ignore this version pattern mychecker-5.6. \n Consider this version pattern mychecker-5.8. \n Some more lines."
-    lines = string.split("\n")
-    checker = MyFakeChecker()
-
-    result1 = checker.get_versions(lines[1], "")
-    assert result1.versions[0] == "UNKNOWN"
-
-    result2 = checker.get_versions(lines[2], "")
-    assert result2.versions[0] == "5.8"
+    def test_multiple_versions(self):
+        string = (
+            "Some other lines. \n"
+            "Consider this version pattern mychecker-5.8. \n"
+            "And this version pattern mychecker-6.0. \n"
+            "Some more lines."
+        )
+        lines = string.split("\n")
+        checker = self.MyFakeChecker()
+
+        result = checker.get_versions(lines[1], "")
+        assert "5.8" in result.versions
+
+        result = checker.get_versions(lines[2], "")
+        assert "6.0" in result.versions
+
+    def test_ignored_versions(self):
+        string = (
+            "Some other lines. \n"
+            "Ignore this version pattern mychecker-5.6. \n"
+            "And this version pattern mychecker-5.6. \n"
+            "Some more lines."
+        )
+        lines = string.split("\n")
+        checker = self.MyFakeChecker()
+
+        result = checker.get_versions(lines[1], "")
+        assert result.versions[0] == "UNKNOWN"
+
+        result = checker.get_versions(lines[2], "")
+        assert result.versions[0] == "UNKNOWN"
+
+    def test_mixed_versions(self):
```

Comment:
Could we modify this to pass all of the lines into the checker at once instead of one line at a time? The point of this test is to see if one result will contain both `5.8` and `6.0`, but not `5.6.`